paddle/pten/core/storage.cc

```diff
@@ -18,7 +18,7 @@ namespace pten {
 
 void TensorStorage::Realloc(size_t size) {
   this->Clear();
-  data_ = paddle::memory::AllocShared(alloc_->place(), size);
+  data_ = alloc_->Allocate(size);
```

Comment:
"Allocator* alloc_" could be nullptr by the time "Realloc()" is called, shall we add validation check here?

Reply:
Thanks for the suggestion. A non-null check should be added here, which I will refine in the next PR.

Reply:
Agreed, I even thought you can leave some comments here for now, since we're at an intermediate state and these codes might experience another refactor in the near future.